doc/proposal/expected/d0323r3.md

```diff
@@ -163,82 +165,266 @@ Using `expected<double, errc>`, the code becomes:
+            return std::make_unexpected(IPv4ParsingError::Failure);
+    }
+    if (items.size() > 1) {
+        for (size_t i = 0; i < items.size() - 1; i++) {
+            if (items[i].value() > 255)
+                return std::make_unexpected(IPv4ParsingError::Failure);
+        }
+    }
+    if (items[items.size() - 1].value() >= pow256(5 - items.size()))
+        return std::make_unexpected(IPv4ParsingError::Failure);
+
+    if (didSeeSyntaxViolation)
+        syntaxViolation(iteratorForSyntaxViolationPosition);
+    for (const auto& item : items) {
+        if (item.value() > 255)
+            syntaxViolation(iteratorForSyntaxViolationPosition);
+    }
+
+    if (items.size() != 4)
+        syntaxViolation(iteratorForSyntaxViolationPosition);
+
+    IPv4Address ipv4 = items.takeLast().value();
+    for (size_t counter = 0; counter < items.size(); ++counter)
+        ipv4 += items[counter].value() * pow256(3 - counter);
+    return ipv4;
+}
+```
+
+Note that the above code uses `std::make_unexpected` because WebKit still uses
+C++14. A C++17 codebase would use deduction guides and `std::unexpected`
+directly.
+
+
 ## Error retrieval and correction
 
-The major advantage of `expected<T,E>` over `optional<T>` is the ability to transport an error, but we didn’t come yet to an example that retrieve the error. First of all, we should wonder what a programmer do when a function call returns an error:
+The major advantage of `expected<T, E>` over `optional<T>` is the ability to
+transport an error. Programmer do the following when a function call returns an
+error:
 
 1. Ignore it.
 2. Delegate the responsibility of error handling to higher layer.
-3. Trying to resolve the error.
+3. Try to resolve the error.
  
-Because the first behavior might lead to buggy application, we won’t consider it in a first time. The handling is dependent of the underlying error type, we consider the `exception_ptr` and the `errc` types.
-
-We spoke about how to use the value contained in the `expected` but didn’t discuss yet the error usage.
+Because the first behavior might lead to buggy application, we ignore the
+usecase. The handling is dependent of the underlying error type, we consider the
+`exception_ptr` and the `errc` types.
 
-A first imperative way to use our error is to simply extract it from the `expected` using the `error()` member function. The following example shows a `divide2` function that return `0` if the error is `divide_by_zero`:
+A first imperative way to use our error is to simply extract it from the
+`expected` using the `error()` member function. The following example shows a
+`divide2` function that return `0` if the error is `divide_by_zero`:
 
 ```c++
-expected<int, errc> divide2(int i, int j)
+int divide2(int i, int j)
 {
-	auto e = safe_divide(i,j);
-	if (!e && e.error().value() == arithmetic_errc::divide_by_zero) {
-		return 0;
-	}
-	return e;
+    auto e = safe_divide(i, j);
+    if (!e)
+        switch (e.error().value()) {
```

Comment:
I would need to fix this `e.error().value()` and use directly `e.error()`. It was already there.